R/parallelization.R

```diff
@@ -1,9 +1,64 @@
 #' Parallelization in purrr
 #'
 #' @description
-#' purrr's map functions have a `.parallel` argument to parallelize a map using
-#' the \CRANpkg{mirai} package. This allows you to run computations in parallel
-#' using more cores on your machine, or distributed over the network.
+#' `r lifecycle::badge("experimental")`
+#'
+#' All map functions allow parallelized operation using \CRANpkg{mirai}. To take
+#' advantage of this, wrap a function that is passed to the `.f` argument of
+#' [map()] or any of its variants with [in_parallel()]. This declares that the
+#' computation should proceed in parallel utilizing multiple cores on your local
+#' machine, or distributed over the network.
+#'
+#' @param .f A fresh formula or function. "Fresh" here means that they should be
+#'   declared in the call to [in_parallel()].
+#' @param ... Arguments to declare in the environment of the function.
+#'
+#' @section How to Use:
+#'
+#' Wrapping a function that is passed to the `.f` argument in [map()] or any of
+#' its variants with [in_parallel()] declares that the map should be performed
+#' in parallel.
+#'
+#' Under the hood, [in_parallel()] provides a systematic way of making `.f`
+#' self-contained so that it can be readily shared with other parallel
+#' processes. It ensures that everything needed by the function is serialized
+#' along with it, but not other objects which happen to be in the function's
+#' enclosing environment. This helps to prevent inadvertently shipping large
+#' data objects when they may not be needed.
+#'
+#' To create self-contained functions:
```

Comment:
This requires a little work because you'll need to:

- Pass along dependencies, such as a list of parameters or metadata, to be packaged up in your self-contained function.
- Fully qualify calls to non-base functions with their namespaces.